src/Text/Pandoc/Readers/Org.hs

```diff
@@ -407,11 +407,21 @@ orgBlock = try $ do
       _         -> withParsed (fmap $ divWithClass blkType)
 
 blockHeaderStart :: OrgParser (Int, String)
-blockHeaderStart = try $ (,) <$> indent <*> blockType
+blockHeaderStart = try $ (,) <$> spaceLength <*> blockType
  where
-  indent    = length      <$> many spaceChar
   blockType = map toLower <$> (stringAnyCase "#+begin_" *> orgArgWord)
 
+spaceLength :: OrgParser Int
+spaceLength = try $ do
+  tabStop  <- getOption readerTabStop
+  spaceStr <- many spaceChar
+  let charLen = \c ->
```

Comment:
I personally prefer to define functions like these in a `where` statement as it's less disrupting and a bit cleaner (IMHO).  The lambda function is not really necessary:

    charLen c = case c of
        ' '  -> 1
        '\t' -> tabStop
        _   -> 0